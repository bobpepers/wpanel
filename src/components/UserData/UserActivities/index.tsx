import { Typography } from '@material-ui/core';
import * as React from 'react';
import { InfoTable } from '../../InfoTable/InfoTable';
import { TableHeaderItemInterface } from '../UserData';

export interface UserActivityProps {
    // tslint:disable:no-any
    classes: any;
    userActivity: any;
    // tslint:enable:no-any
    rows: TableHeaderItemInterface[];
    page: number;
    rowsPerPage: number;
    total: number;
    handleChangePage: (page: number) => void;
    handleChangeRowsPerPage: (rows: number) => void;
}

export class UserActivities extends React.Component<UserActivityProps> {
    public render() {
        const {
            userActivity,
            classes,
            rowsPerPage,
            page,
            rows,
            handleChangePage,
            handleChangeRowsPerPage,
            total,
        } = this.props;
        const content = userActivity && userActivity.length ? (
            <InfoTable
                dataLength={total}
                rows={rows}
                data={userActivity}
                page={page}
                rowsPerPage={rowsPerPage}
                handleChangePage={handleChangePage}
                handleChangeRowsPerPage={handleChangeRowsPerPage}
                hidePagination={false}
            />) : (
                <Typography variant="h6" gutterBottom={true} style={{ color: '#757575', paddingLeft: 26, paddingBottom: 26 }}>
                    No user activities
            </Typography>
            );

        return (
            <React.Fragment>
                <Typography variant="h6" gutterBottom={true} className={classes.title}>
                    User activities
                </Typography>
                {content}
            </React.Fragment>
        );
    }
}
