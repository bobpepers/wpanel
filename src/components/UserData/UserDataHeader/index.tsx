import {
    Grid,
    Typography,
} from '@material-ui/core';
import { KeyboardArrowRight } from '@material-ui/icons';
import * as React from 'react';

export interface UserDataHeaderProps {
    // tslint:disable-next-line:no-any
    classes: any;
    user: string;
    // tslint:disable-next-line:no-any
    goBack: (event: any) => void;
    pathname: string;
}

export class UserDataHeader extends React.Component<UserDataHeaderProps> {
    public render() {
        const { classes, user, pathname } = this.props;

        const path = pathname.includes('/tower/users') ?
            'User Directory' :
            pathname.includes('/tower/pending') ?
            'Pending Documents' :
            pathname.includes('/tower/admin-activities') ?
            'Admin Activities' :
            'User Activities';

        return (
            <Grid container={true} className={classes.header}>
                <Typography variant="subheading" onClick={this.props.goBack} className={classes.link}>
                    {path}
                </Typography>
                <KeyboardArrowRight className={classes.arrow}/>
                <Typography variant="subheading" style={{ letterSpacing: '0.44px' }}>
                    {user}
                </Typography>
            </Grid>
        );
    }
}
