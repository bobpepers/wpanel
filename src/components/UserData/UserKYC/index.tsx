import {
    createStyles,
    Grid,
    Switch,
    Theme,
    Typography,
    WithStyles,
    withStyles,
} from '@material-ui/core';
import * as React from 'react';
import { localeDate } from '../../../helpers';
import { icons } from './icons';

const styles = (theme: Theme) => createStyles({
    paper: {
        padding: '20px 24px 12px 0px',
    },
    formControl: {
        marginBottom: theme.spacing.unit * 3,
    },
    gridRow: {
        justifyContent: 'space-between',
        paddingRight: theme.spacing.unit * 3,
    },
    gridColumn: {
        display: 'flex',
        flexDirection: 'column',
    },
    gridRowWithDivider: {
        borderColor: '#e8e8e8',
        border: 0,
        borderLeft: 1,
        borderStyle: 'solid',
        justifyContent: 'space-between',
        marginTop: theme.spacing.unit * 1.5,
        marginBottom: theme.spacing.unit * 1.5,
        height: '15px',
    },
    title: {
        marginBottom: theme.spacing.unit * 3,
        marginLeft: theme.spacing.unit * 3,
        letterSpacing: '0.15px',
        fontWeight: 600,
    },
    info: {
        opacity: 0.72,
        letterSpacing: '0.25px',
        marginBottom: theme.spacing.unit * 2,
    },
    level: {
        display: 'flex',
        alignItems: 'flex-start',
        justifyContent: 'space-between',
    },
});

interface StyleProps extends WithStyles<typeof styles> {
    theme?: Theme;
}

// tslint:disable:no-any
export interface UserKYCProps {
    user: any;
    editLabel: (key: string, value: string, scope: string) => void;
    alertPush: ({ message, type }) => void;
}
// tslint:enable:no-any

type Props = StyleProps & UserKYCProps;

class UserKYCComponent extends React.Component<Props> {
    public render() {
        const {
            classes,
        } = this.props;

        return (
            <div className={classes.paper}>
                <Grid container={true}>
                    <Grid item={true} xs={12} className={classes.gridRow}>
                        <Typography variant="h6" className={classes.title}>KYC verification</Typography>
                    </Grid>

                    <Grid item={true} xs={1} className={classes.gridColumn} style={{ alignItems: 'center', marginTop: 10 }}>
                        {this.checkLabel('email') ? icons('success') : icons('email')}
                        <div className={classes.gridRowWithDivider} />
                        {this.checkLabel('phone') ? icons('success') : icons('phone')}
                        <div className={classes.gridRowWithDivider} />
                        {this.checkLabel('document') ? icons('success') : icons('document')}
                    </Grid>
                    <Grid item={true} container={true} xs={11} className={classes.gridColumn}>
                        {['Email', 'Phone', 'Document'].map((item, index) => this.getRow(item, index))}
                    </Grid>
                </Grid>
            </div>
        );
    }

    private checkLabel = (key: string) => {
        const { user } = this.props;
        return user.labels.find(label => label.key === key && label.value === 'verified');
    }

    private getRow = (key: string, index: number) => {
        const { user, classes } = this.props;
        const status = this.checkLabel(key.toLowerCase()) ? 'verified' : 'verification';

        return (
            <Grid item={true} className={classes.level} key={key}>
                <Grid item={true} className={classes.gridColumn} style={{ alignItems: 'flex-start' }} xs={11}>
                    <Typography variant="subtitle1" style={{ letterSpacing: '0.44px' }}>
                        {key} {status}
                    </Typography>
                    <Typography variant="body2" className={classes.info}>
                        {status === 'verified' ? `Level ${index + 1} completed at ${localeDate(user.labels.find(label => label.key === key.toLowerCase()).updated_at, 'shortDate')}` : `Confirm the ${key}`}
                    </Typography>
                </Grid>
                <Grid item={true} className={classes.gridColumn}>
                    <Switch
                        checked={status === 'verified'}
                        onChange={this.onClick(key.toLowerCase())}
                        color="primary"
                    />
                </Grid>
            </Grid>
        );
    };

    private onClick = (key: string) => () => {
        const { user } = this.props;
        const label = user.labels.find(item => item.key === key);
        if (!label) {
            this.props.alertPush({
                message: ['Label doesn\'t exist'],
                type: 'error',
            });
        } else if (label.value === 'verified') {
            this.props.editLabel(key, 'rejected', label.scope);
        } else {
            this.props.editLabel(key, 'verified', label.scope);
        }
    };
}

export const UserKYC = withStyles(styles)(UserKYCComponent);
