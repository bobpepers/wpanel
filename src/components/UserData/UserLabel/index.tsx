import {
    Button,
    Grid,
    Typography,
} from '@material-ui/core';
import SvgIcon from '@material-ui/core/SvgIcon';
import { AddCircleOutline } from '@material-ui/icons';
import * as React from 'react';
import { EditLabel } from '../../';

export interface UserLabelProps {
    // tslint:disable-next-line:no-any
    classes: any;
    // tslint:disable-next-line:no-any
    user: any;
    newLabelName: string;
    newLabelValue: string;
    newLabelScope: string;
    addNewLabel: () => void;
    editLabel: (key: string, value: string, scope: string) => void;
    deleteUserLabel: (uid: string, key: string, scope: string) => void;
    changeLabelName: (value: string) => void;
    changeLabelScope: (value: string) => void;
    changeLabelValue: (value: string) => void;
    isAddLabelModalOpened: boolean;
    isEditLabelModalOpened: boolean;
    openAddLabelModal: () => void;
    openEditLabelModal: (key: string, value: string, scope: string) => void;
    closeModal: () => void;
}

export class UserLabel extends React.Component<UserLabelProps> {
    public render() {
        const {
            classes,
            user,
            newLabelName,
            newLabelValue,
            newLabelScope,
        } = this.props;

        return (
            <div className={classes.paper}>
                <Grid item={true} xs={12} style={{ display: 'flex', justifyContent: 'space-between'}}>
                    <Typography variant="h6" gutterBottom={true} className={classes.title} style={{ paddingLeft: 0, paddingTop: 0 }}>
                        Labels
                    </Typography>
                    <Grid item={true}>
                        <Button onClick={e => this.openAddLabelModal()} disableFocusRipple={true}>
                            <AddCircleOutline style={{ color: '#979797', opacity: 0.6 }} />
                        </Button>
                    </Grid>
                </Grid>
                <Grid container={true} justify={'flex-start'} spacing={16}>
                {user && user.labels && user.labels.length === 0 ? (
                    <Typography variant="h6" gutterBottom={true} style={{ color: '#757575', marginLeft: 8 }}>
                        No labels
                    </Typography>
                    ) : this.renderContent()}
                    <EditLabel
                        editLabel={this.props.addNewLabel}
                        handleChangeLabelName={this.props.changeLabelName}
                        handleChangeLabelScope={this.props.changeLabelScope}
                        handleChangeLabelValue={this.props.changeLabelValue}
                        modalClose={this.modalClose}
                        name={newLabelName}
                        open={this.props.isAddLabelModalOpened}
                        scope={newLabelScope}
                        value={newLabelValue}
                    />
                    <EditLabel
                        editLabel={this.props.editLabel}
                        handleChangeLabelName={this.props.changeLabelName}
                        handleChangeLabelScope={this.props.changeLabelScope}
                        handleChangeLabelValue={this.props.changeLabelValue}
                        modalClose={this.modalClose}
                        name={newLabelName}
                        open={this.props.isEditLabelModalOpened}
                        scope={newLabelScope}
                        value={newLabelValue}
                    />
                </Grid>

            </div>
        );
    }

    // tslint:disable-next-line:no-any
    private renderContent = () => this.props.user.labels.map((label: any, i: number) => this.getLabelData(label, i, this.props.classes));

    // tslint:disable-next-line:no-any
    private getLabelData = (label: any, i: number, classes: any) => {
        const { user } = this.props;
        switch (label.key) {
            case 'email':
                return (
                    <Grid item={true} key={i}>
                        <Grid
                            container={true}
                            justify={'space-between'}
                            className={classes.label}
                            style={{ backgroundColor: '#43A047' }}
                        >
                            <Typography
                                onClick={e => this.openEditLabelModal(label.key, label.value, label.scope)}
                                className={classes.labelName}
                            >
                                email:{label.value}
                            </Typography>
                            <SvgIcon
                                onClick={e => this.deleteLabel(user.uid, label.key, label.scope)}
                                className={classes.icon}
                                viewBox="0 0 32 32"
                            >
                                <path
                                    d="m15,0.25c8.15675,0 14.75,6.59325 14.75,14.75c0,8.15675 -6.59325,14.75 -14.75,14.75c-8.15675,0 -14.75,-6.59325 -14.75,-14.75c0,-8.15675 6.59325,-14.75 14.75,-14.75m5.29525,7.375l-5.29525,5.29525l-5.29525,-5.29525l-2.07975,2.07975l5.29525,5.29525l-5.29525,5.29525l2.07975,2.07975l5.29525,-5.29525l5.29525,5.29525l2.07975,-2.07975l-5.29525,-5.29525l5.29525,-5.29525l-2.07975,-2.07975z"
                                    id="svg_1"
                                    fill="#ffffff"
                                    stroke="null"
                                />
                            </SvgIcon>
                        </Grid>
                    </Grid>
                );
            case 'phone':
                return (
                    <Grid item={true} key={i}>
                        <Grid
                            container={true}
                            justify={'space-between'}
                            className={classes.label}
                            style={{ backgroundColor: '#309CEA' }}
                        >
                            <Typography
                                onClick={e => this.openEditLabelModal(label.key, label.value, label.scope)}
                                className={classes.labelName}
                            >
                                phone:{label.value}
                            </Typography>
                            <SvgIcon
                                onClick={e => this.deleteLabel(user.uid, label.key, label.scope)}
                                className={classes.icon}
                                viewBox="0 0 32 32"
                            >
                                <path
                                    d="m15,0.25c8.15675,0 14.75,6.59325 14.75,14.75c0,8.15675 -6.59325,14.75 -14.75,14.75c-8.15675,0 -14.75,-6.59325 -14.75,-14.75c0,-8.15675 6.59325,-14.75 14.75,-14.75m5.29525,7.375l-5.29525,5.29525l-5.29525,-5.29525l-2.07975,2.07975l5.29525,5.29525l-5.29525,5.29525l2.07975,2.07975l5.29525,-5.29525l5.29525,5.29525l2.07975,-2.07975l-5.29525,-5.29525l5.29525,-5.29525l-2.07975,-2.07975z"
                                    id="svg_1"
                                    fill="#ffffff"
                                    stroke="null"
                                />
                            </SvgIcon>
                        </Grid>
                    </Grid>
                );
            case 'document':
                return (
                    <Grid item={true} key={i}>
                        <Grid
                            container={true}
                            justify={'space-between'}
                            className={classes.label}
                            style={{ backgroundColor: '#3F51B5' }}
                        >
                            <Typography
                                onClick={e => this.openEditLabelModal(label.key, label.value, label.scope)}
                                className={classes.labelName}
                            >
                                document:{label.value}
                            </Typography>
                            <SvgIcon
                                onClick={e => this.deleteLabel(user.uid, label.key, label.scope)}
                                className={classes.icon}
                                viewBox="0 0 32 32"
                            >
                                <path
                                    d="m15,0.25c8.15675,0 14.75,6.59325 14.75,14.75c0,8.15675 -6.59325,14.75 -14.75,14.75c-8.15675,0 -14.75,-6.59325 -14.75,-14.75c0,-8.15675 6.59325,-14.75 14.75,-14.75m5.29525,7.375l-5.29525,5.29525l-5.29525,-5.29525l-2.07975,2.07975l5.29525,5.29525l-5.29525,5.29525l2.07975,2.07975l5.29525,-5.29525l5.29525,5.29525l2.07975,-2.07975l-5.29525,-5.29525l5.29525,-5.29525l-2.07975,-2.07975z"
                                    id="svg_1"
                                    fill="#ffffff"
                                    stroke="null"
                                />
                            </SvgIcon>
                        </Grid>
                    </Grid>
                );
            default:
                return (
                    <Grid item={true} key={i}>
                        <Grid
                            container={true}
                            justify={'space-between'}
                            className={classes.label}
                            style={{ backgroundColor: '#e0e0e0' }}
                        >
                            <Typography
                                onClick={e => this.openEditLabelModal(label.key, label.value, label.scope)}
                                style={{ paddingTop: 5, color: '#757575', fontSize: 14, marginRight: 7 }}
                            >
                                {label.key}
                            </Typography>
                            <SvgIcon
                                onClick={e => this.deleteLabel(user.uid, label.key, label.scope)}
                                className={classes.icon}
                                viewBox="0 0 32 32"
                            >
                                <path
                                    d="m15,0.25c8.15675,0 14.75,6.59325 14.75,14.75c0,8.15675 -6.59325,14.75 -14.75,14.75c-8.15675,0 -14.75,-6.59325 -14.75,-14.75c0,-8.15675 6.59325,-14.75 14.75,-14.75m5.29525,7.375l-5.29525,5.29525l-5.29525,-5.29525l-2.07975,2.07975l5.29525,5.29525l-5.29525,5.29525l2.07975,2.07975l5.29525,-5.29525l5.29525,5.29525l2.07975,-2.07975l-5.29525,-5.29525l5.29525,-5.29525l-2.07975,-2.07975z"
                                    id="svg_1"
                                    fill="#979797"
                                    stroke="null"
                                />
                            </SvgIcon>
                        </Grid>
                    </Grid>
                );
        }
    };

    private openAddLabelModal = () => {
        this.props.openAddLabelModal();
    };

    private openEditLabelModal = (key: string, value: string, scope: string) => {
        this.props.openEditLabelModal(key, value, scope);
    };

    private modalClose = () => {
        this.props.closeModal();
    };

    private deleteLabel = (uid: string, key: string, scope: string) => {
        this.props.deleteUserLabel(uid, key, scope);
    };
}
