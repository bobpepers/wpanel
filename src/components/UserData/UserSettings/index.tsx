import {
    createStyles,
    FormControl,
    Grid,
    Input,
    InputLabel,
    MenuItem,
    Select,
    Switch,
    Theme,
    Typography,
    WithStyles,
    withStyles,
} from '@material-ui/core';
import * as React from 'react';
import { CurrentUserInterface } from '../../../modules';

const styles = (theme: Theme) => createStyles({
    paper: {
        padding: '20px 24px 12px 24px',
    },
    formControl: {
        marginBottom: theme.spacing.unit * 3,
    },
    gridRow: {
        justifyContent: 'space-between',
        paddingRight: theme.spacing.unit * 5,
    },
    gridRowCenter: {
        display: 'flex',
        alignItems: 'center',
    },
    title: {
        marginBottom: theme.spacing.unit * 3,
        letterSpacing: '0.15px',
        fontWeight: 600,
    },
    subtitle: {
        color: '#757575',
        marginBottom: theme.spacing.unit,
    },
});

interface StyleProps extends WithStyles<typeof styles> {
    theme?: Theme;
}

// tslint:disable:no-any
export interface UserSettingsProps {
    user: any;
    handleChangeUserState: (e: any) => void;
    handleChangeRole: (e: any) => void;
    handleChangeUserOTP: (e: any) => void;
    currentUser: CurrentUserInterface;
}
// tslint:enable:no-any

const stateTypes = [
    {
        value: 'Pending',
        key: 'pending',
    },
    {
        value: 'Active',
        key: 'active',
    },
    {
        value: 'Banned',
        key: 'banned',
    },
];

const roleTypes = [
    {
        value: 'Admin',
        key: 'admin',
    },
    {
        value: 'Member',
        key: 'member',
    },
    {
        value: 'Super Admin',
        key: 'superadmin',
    },
    {
        value: 'Accountant',
        key: 'accountant',
    },
    {
        value: 'Compliance',
        key: 'compliance',
    },
    {
        value: 'Technical',
        key: 'technical',
    },
    {
        value: 'Support',
        key: 'support',
    },
    {
        value: 'Trader',
        key: 'trader',
    },
    {
        value: 'Broker',
        key: 'broker',
    },
];

type Props = StyleProps & UserSettingsProps;

class UserSettingsComponent extends React.Component<Props> {
    public render() {
        const {
            classes,
            user,
            handleChangeUserState,
            handleChangeRole,
            handleChangeUserOTP,
            currentUser,
        } = this.props;

        return (
            <div className={classes.paper}>
                <Grid container={true}>
                    <Grid item={true} xs={6} className={classes.gridRow}>
                        <Typography variant="h6" className={classes.title}>Settings</Typography>
                        <Grid container={true} item={true} justify={'flex-start'} xs={12}>
                            <FormControl fullWidth={true} className={classes.formControl}>
                                <InputLabel htmlFor="state-label-placeholder">State</InputLabel>
                                <Select
                                    value={user.state}
                                    onChange={handleChangeUserState}
                                    input={<Input id="state-label-placeholder" />}
                                >
                                    {stateTypes.map(option => <MenuItem key={option.key} value={option.key}>{option.value}</MenuItem>)}
                                </Select>
                            </FormControl>
                        </Grid>
                        <Grid container={true} item={true} justify={'flex-start'} xs={12}>
                            {currentUser && currentUser.role === 'superadmin' &&
                            <FormControl fullWidth={true} className={classes.formControl}>
                                <InputLabel htmlFor="role-label-placeholder">Role</InputLabel>
                                <Select
                                    value={user.role}
                                    onChange={handleChangeRole}
                                    input={<Input id="role-label-placeholder"/>}
                                >
                                    {roleTypes.sort((a, b) => a.value.localeCompare(b.value)).map(option =>
                                        <MenuItem key={option.key} value={option.key}>{option.value}</MenuItem>)}
                                </Select>
                            </FormControl>}
                        </Grid>
                    </Grid>
                    <Grid item={true} xs={6} className={classes.gridRowCenter}>
                        <Grid container={true} alignItems={'center'}>
                            <Grid item={true} xs={12}>
                                <Typography variant="caption" gutterBottom={true}>Authorization 2FA</Typography>
                            </Grid>
                            <Grid container={true} item={true} xs={6} justify={'flex-start'} alignItems="center">
                                <Typography>{user.otp ? 'Enabled' : 'Disabled'}</Typography>
                            </Grid>
                        </Grid>
                        <Grid container={true} item={true} xs={6} justify="flex-end" alignItems="center">
                            <Switch
                                checked={user.otp}
                                onChange={handleChangeUserOTP}
                                color="primary"
                            />
                        </Grid>
                    </Grid>
                </Grid>
            </div>
        );
    }
}

export const UserSettings = withStyles(styles)(UserSettingsComponent);
