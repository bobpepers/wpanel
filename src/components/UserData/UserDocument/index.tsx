import { Typography } from '@material-ui/core';
import * as React from 'react';
import { InfoTable } from '../../InfoTable/InfoTable';
import { TableHeaderItemInterface } from '../UserData';

export interface UserDocumentProps {
    // tslint:disable:no-any
    user: any;
    classes: any;
    // tslint:enable:no-any
    documentsRows: TableHeaderItemInterface[];
    handleOpenDocumentCarousel: (index: number) => void;
}

export class UserDocument extends React.Component<UserDocumentProps> {
    public render() {
        const {user, classes, handleOpenDocumentCarousel} = this.props;
        const content = user && user.documents && user.documents.length ? (
            <InfoTable
                dataLength={this.props.documentsRows.length}
                rows={this.props.documentsRows}
                data={user.documents}
                rowsPerPage={user.documents.length}
                hidePagination={true}
                handleOpen={handleOpenDocumentCarousel}
            />) : (
            <Typography variant="h6" gutterBottom={true} style={{ color: '#757575', paddingLeft: 26, paddingBottom: 26}}>
                No documents
            </Typography>
        );

        return (
            <React.Fragment>
                <Typography variant="h6" gutterBottom={true} className={classes.title}>
                    Documents
                </Typography>
                {content}
            </React.Fragment>
        );
    }
}
