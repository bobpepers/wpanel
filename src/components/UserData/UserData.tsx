import {
    createStyles,
    Grid,
    Paper,
    Theme,
    WithStyles,
    withStyles,
} from '@material-ui/core';
import * as React from 'react';
import { DocumentCarousel } from '../';
import { CurrentUserInterface } from '../../modules';
import { UserActivities } from './UserActivities';
import { UserDataHeader } from './UserDataHeader';
import { UserDocument } from './UserDocument';
import { UserKYC } from './UserKYC';
import { UserLabel } from './UserLabel';
import { UserSettings } from './UserSettings';
import { UserSummary } from './UserSummary';

export interface TableHeaderItemInterface {
    key: string;
    alignRight: boolean;
    label: string;
}

export interface UserDataProps {
    addNewLabel: () => void;
    editLabel: (key: string, value: string, scope: string) => void;
    changeLabelName: (value: string) => void;
    changeLabelScope: (value: string) => void;
    changeLabelValue: (value: string) => void;
    closeModal: () => void;
    deleteUserLabel: (uid: string, key: string, scope: string) => void;
    // tslint:disable-next-line:no-any
    handleChangeUserState: (e: any) => void;
    // tslint:disable-next-line:no-any
    handleChangeRole: (e: any) => void;
    // tslint:disable-next-line:no-any
    handleChangeUserOTP: (e: any) => void;
    newLabelName: string;
    newLabelScope: string;
    newLabelValue: string;
    isAddLabelModalOpened: boolean;
    isEditLabelModalOpened: boolean;
    openAddLabelModal: () => void;
    openEditLabelModal: (key: string, value: string, scope: string) => void;
    // tslint:disable-next-line:no-any
    user: any;
    documentsRows: TableHeaderItemInterface[];
    handleEditLabel: (key: string, value: string, scope: string) => void;
    activityRows: TableHeaderItemInterface[];
    // tslint:disable-next-line:no-any
    userActivity: any;
    page: number;
    rowsPerPage: number;
    total: number;
    handleChangePage: (page: number) => void;
    handleChangeRowsPerPage: (rows: number) => void;
    // tslint:disable-next-line:no-any
    goBack: (event: any) => void;
    pathname: string;
    currentUser: CurrentUserInterface;
    alertPush: ({type, message}) => void;
    handleOpenDocumentCarousel: (index: number) => void;
    handleCloseDocumentCarousel: () => void;
    openDocumentCarousel: boolean;
    documentIndex: number;
    handleNavigateDocumentCarousel: (index: number) => void;
}

const styles = (theme: Theme) => createStyles({
    textField: {
        marginLeft: theme.spacing.unit,
        marginRight: theme.spacing.unit,
        width: '75%',
    },
    menu: {
        width: 200,
    },
    link: {
        cursor: 'pointer',
        textDecoration: 'none',
        color: '#3598D5',
        fontSize: '16px',
        letterSpacing: '0.44px',
    },
    arrow: {
        color: '#979797',
        paddingTop: '3px',
        margin: '0 10px',
    },
    header: {
        display: 'flex',
        alignItems: 'center',
        margin: '0 24px 24px 0',
        fontWeight: 600,
    },
    label: {
        height: 32,
        paddingLeft: 16,
        borderRadius: 24,
        width: 'auto',
        cursor: 'pointer',
        letterSpacing: '0.15px',
        fontWeight: 600,
    },
    icon: {
        width: 20,
        height: 20,
        margin: '7px 4px',
        cursor: 'pointer',
        opacity: 0.6,
    },
    labelName: {
        paddingTop: 5,
        color: '#ffffff',
        fontSize: 14,
        marginRight: 7,
        letterSpacing: '0.25px',
    },
    paper: {
        padding: '20px 24px 24px 24px',
    },
    title: {
        marginBottom: theme.spacing.unit * 3,
        letterSpacing: '0.15px',
        fontWeight: 600,
        paddingLeft: 26,
        paddingTop: 26,
    },
});

interface StyleProps extends WithStyles<typeof styles> {
    theme?: Theme;
}

type Props = StyleProps & UserDataProps;

class UserDataComponent extends React.Component<Props> {

    public render() {
        const {
            classes,
            user,
            newLabelName,
            newLabelValue,
            newLabelScope,
            addNewLabel,
            editLabel,
            deleteUserLabel,
            changeLabelName,
            changeLabelScope,
            changeLabelValue,
            isAddLabelModalOpened,
            isEditLabelModalOpened,
            openAddLabelModal,
            openEditLabelModal,
            closeModal,
            handleChangeUserState,
            handleChangeRole,
            handleChangeUserOTP,
            documentsRows,
            handleEditLabel,
            userActivity,
            activityRows,
            page,
            rowsPerPage,
            handleChangePage,
            handleChangeRowsPerPage,
            total,
            goBack,
            pathname,
            currentUser,
            alertPush,
            handleCloseDocumentCarousel,
            handleOpenDocumentCarousel,
            openDocumentCarousel,
            documentIndex,
            handleNavigateDocumentCarousel,
        } = this.props;

        return (
            <div className="user-data">
                <UserDataHeader classes={classes} user={user.email} goBack={goBack} pathname={pathname}/>
                <Grid container={true} spacing={24} direction={'row'}>
                    <Grid item={true} xs={12} lg={6}>
                        <Paper style={{padding: 20}}>
                            <UserSummary
                                user={user}
                            />
                        </Paper>
                    </Grid>
                    <Grid item={true} xs={12} lg={6}>
                        <Paper style={{ marginBottom: 24 }}>
                            <UserKYC
                                user={user}
                                editLabel={handleEditLabel}
                                alertPush={alertPush}
                            />
                        </Paper>
                        <Paper style={{marginBottom: 24}}>
                            <UserSettings
                                user={user}
                                handleChangeUserState={handleChangeUserState}
                                handleChangeRole={handleChangeRole}
                                handleChangeUserOTP={handleChangeUserOTP}
                                currentUser={currentUser}
                            />
                        </Paper>
                    </Grid>
                    <Grid item={true} xs={12} lg={12}>
                        <Paper>
                            <UserLabel
                                user={user}
                                newLabelName={newLabelName}
                                newLabelValue={newLabelValue}
                                newLabelScope={newLabelScope}
                                addNewLabel={addNewLabel}
                                editLabel={editLabel}
                                deleteUserLabel={deleteUserLabel}
                                changeLabelName={changeLabelName}
                                changeLabelScope={changeLabelScope}
                                changeLabelValue={changeLabelValue}
                                isAddLabelModalOpened={isAddLabelModalOpened}
                                isEditLabelModalOpened={isEditLabelModalOpened}
                                openAddLabelModal={openAddLabelModal}
                                openEditLabelModal={openEditLabelModal}
                                closeModal={closeModal}
                                classes={classes}
                            />
                        </Paper>
                    </Grid>
                    <Grid item={true} xs={12} lg={12}>
                        <Paper style={{ marginBottom: 15 }}>
                            <UserDocument
                                user={user}
                                documentsRows={documentsRows}
                                classes={classes}
                                handleOpenDocumentCarousel={handleOpenDocumentCarousel}
                            />
                            {openDocumentCarousel &&
                                <DocumentCarousel
                                    documents={user.documents}
                                    documentIndex={documentIndex}
                                    handleClose={handleCloseDocumentCarousel}
                                    handleNavigate={handleNavigateDocumentCarousel}
                                />
                            }
                        </Paper>
                    </Grid>
                    <Grid item={true} xs={12} lg={12}>
                        <Paper style={{ marginBottom: 15 }}>
                            <UserActivities
                                userActivity={userActivity}
                                page={page}
                                rowsPerPage={rowsPerPage}
                                handleChangePage={handleChangePage}
                                handleChangeRowsPerPage={handleChangeRowsPerPage}
                                rows={activityRows}
                                classes={classes}
                                total={total}
                            />
                        </Paper>
                    </Grid>
                </Grid>
            </div>
        );
    }

}

export const UserData = withStyles(styles)(UserDataComponent);
