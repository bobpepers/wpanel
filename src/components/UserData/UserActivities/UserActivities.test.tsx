import {shallow} from 'enzyme';
import * as React from 'react';
import { UserActivities, UserActivityProps } from '.';
import { TableHeaderItemInterface } from '../UserData';

const defaults: UserActivityProps = {
    rows: {} as TableHeaderItemInterface[],
    userActivity: {},
    classes: '',
    page: 0,
    rowsPerPage: 0,
    total: 0,
    handleChangePage: jest.fn(),
    handleChangeRowsPerPage: jest.fn(),
};

describe('UserActivities component', () => {

    it('should render', () => {
        const wrapper = shallow(<UserActivities {...defaults} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toBeDefined();
    });
});
