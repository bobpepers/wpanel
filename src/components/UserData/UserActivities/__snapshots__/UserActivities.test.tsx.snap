// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserActivities component should render 1`] = `
<Fragment>
  <WithStyles(Typography)
    gutterBottom={true}
    variant="h6"
  >
    User activities
  </WithStyles(Typography)>
  <WithStyles(Typography)
    gutterBottom={true}
    style={
      Object {
        "color": "#757575",
        "paddingBottom": 26,
        "paddingLeft": 26,
      }
    }
    variant="h6"
  >
    No user activities
  </WithStyles(Typography)>
</Fragment>
`;
